Updates to the 6.4 release notes (#10097)
apple/swift-package-manager: Sources/PackageManagerDocs/Documentation.docc/ReleaseNotes/6.4.md

# SwiftPM 6.4 Release Notes

Swift Package Manager 6.4 makes Swift Build the new default build system, providing a unified build experience across all platforms Swift supports. This release also adds first-class support for generating Software Bill of Materials (SBOM) documents.

## SwiftBuild: The new default build system

SwiftPM has changed its default build system to Swift Build. Read more [here](<doc:SwiftBuildDefault>).

> ### Key changes from the native build system

Reviewer: Most users haven't seen the 'native' terminology before, I wonder if we should just refer to it as the 'previous' build system here

- Build artifacts are output to a different location. For all build systems, using `swift build --show-bin-path <other build arguments>` is the recommended way to determine the build output location.
- **`--static-swift-stdlib`**: Stricter validation (errors instead of silently ignoring)
  - The native build system silently ignored this option on some platforms; Swift Build now produces an error.

>   - Swift Build generates an error on platforms where static libraries are not supported.

Reviewer: ```suggestion
  - Swift Build generates an error on platforms where statically linking the standard library is not supported.
```

> - Swift Build does not support test targets depending on other test targets. To work around this,

Reviewer: It the other PR ends up landing we can drop this bullet point

- Swift Build fails to build projects with Linux SDKs generated _by_ the SDK generator.
  - **Tracking**: [package-manager#10006](https://github.com/swiftlang/swift-package-manager/issues/10006)

- Using the `--explicit-target-dependency-import-check` command line option does not behave as expected when building with Swift Build.
  - **Tracking**: [swiftlang/swift-package-manager#9620](https://github.com/swiftlang/swift-package-manager/issues/9620)

- Swift Build does not support overlapping executable product names and library product names (case-insensitive).
  - **Tracking**: [swiftlang/swift-package-manager#9184](https://github.com/swiftlang/swift-package-manager/issues/9184)

> - `swift sdk configure` was recently updated for the native build system. The functionality is being migrated to Swift Build.

Reviewer: The old and new build systems have equivalent functionality yet so I think this bullet point can just be dropped
